server/src/test/java/io/crate/analyze/DeleteAnalyzerTest.java

```diff
@@ -82,7 +82,7 @@ public void testDeleteWhereSysColumn() throws Exception {
 
     @Test
     public void testDeleteWherePartitionedByColumn() throws Exception {
-        AnalyzedDeleteStatement delete = e.analyze("delete from parted where date = 1395874800000");
+        AnalyzedDeleteStatement delete = e.analyze("delete from parted where date = 1395874800000::timestamp");
```

Comment:
Would `date` be casted to `long` without the explicit timestamp cast?

How would something like `date >= '2019-01-01'` behave?

Reply:
> Would date be casted to long without the explicit timestamp cast?
Yes. But the cast optimizer rules would change the cast to `1395874800000::timestamp` eventually. I just wanted to avoid changing the assertions in that analyzer test.

> How would something like date >= '2019-01-01' behave?
The same, cast on date will be replaced afterwards by the cast optimizer rules.

Reply:
I think `date::text >= '2019-01-01'` has different semantics than `date >= '2019-01-01'::timestamp` has. 

If we "restore" the later semantics only by the optimization, it's no longer only a performance optimization, but will be required for correct behavior - which I think is bad.

For the numeric cases `x::int = 1` or `x = ::long` have the same behavior, so I think there it doesn't matter and is okay.

Maybe we should add additional `timestamp, text` overloads for the comparison operators?

Reply:
> I think date::text >= '2019-01-01' has different semantics than date >= '2019-01-01'::timestamp has.

Yeah right, good point.

> Maybe we should add additional timestamp, text overloads for the comparison operators?

Yes good idea. Could we then postpone this until the operators are refactored to use the new signature based function registry which easily allow overloads? 
I think adding this overload to the existing (old/deprecated) function resolver maybe tricky.

Reply:
> Could we then postpone this until the operators are refactored to use the new signature based function registry which easily allow overloads?

yep